h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -205,23 +208,39 @@ public void onBuildFailure(final ParallelModelBuilder.ModelBuildFailure modelBui
       } finally {
         parallelSearchGridLock.unlock();
       }
+    }
+
+    @Override
+    public void postBuildSuccess(Model finishedModel, ParallelModelBuilder parallelModelBuilder) {
+      attemptBuildNextModel(parallelModelBuilder, finishedModel);
+    }
+
+    @Override
+    public void postBuildFailure(ParallelModelBuilder.ModelBuildFailure modelBuildFailure, ParallelModelBuilder parallelModelBuilder) {
       attemptBuildNextModel(parallelModelBuilder, null);
     }
 
     private void attemptBuildNextModel(final ParallelModelBuilder parallelModelBuilder, final Model previousModel) {
-      // Attempt to train next model
-      final MP nextModelParams = getNextModelParams(hyperspaceIterator, previousModel, grid);
-      if (nextModelParams != null
-              && isThereEnoughTime()
-              && !_job.stop_requested()
-              && !_hyperSpaceWalker.stopEarly(previousModel, grid.getScoringInfos())) {
 
-        // FIXME: ModelFeeder's <MP> type parameter is hiding GridSearch's one. Ideally would have moved ModelFeeder outside but it is too coupled with GridSearch class.
-        reconcileMaxRuntime(grid._key, hyperspaceIterator, nextModelParams);
+      synchronized (hyperspaceIterator) {
+        int numberOfExpectedToCompleteModels = parallelModelBuilder.getModelInProgressCounter().get();
```

Comment:
please keep the `inProgress` name here. This just adds confusion as we don't know when those models are expected to complete... like total of expected models to complete? current expected models to complete? especially hen reading the boolean logic below.

Reply:
@seb-h2o I see your point. Agree

Reply:
@seb-h2o fixed